### PR TITLE
docs: add Nomad Autoscaler HA configuration details.

### PR DESCRIPTION
**Pull request by jrasell**

### CODE REVIEW

**Review by Juanadelacuesta — APPROVED** [review]
great work! Would it be interesting to add a link to the locks and SDK in case people want to see the internals?

File: website/content/tools/autoscaling/agent/high_availability.mdx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## `high_availability` Parameters
+
+- `enabled` `(bool: false)` - Specifies whether the Nomad Autoscaler agent will
+  start in high availability mode. If enabled, the agent instance attempts to
+  hold a lock over a Nomad variable and will only execute if the lock is
+  successfully acquired.
+
+- `lock_path` `(string: "nomad-autoscaler/lock")` - Defines the path of the Nomad
+  variable that will be used to sync the leader when running on high
```
Comment (pkazmierczak) [inline website/content/tools/autoscaling/agent/high_availability.mdx:37]:
```suggestion
  variable that will be used to sync the leader when running in high
```

File: website/content/tools/autoscaling/agent/high_availability.mdx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  availability mode. This parameter needs to be the same on all Nomad Autoscaler
+  agents expected to take part in the same leadership election process.
+
+- `lock_ttl` `(string: "60s")` - Defines the lease period or TTL of the lock used
+  to sync the leader when running on high availability mode. If the Nomad Autoscaler
```
Comment (pkazmierczak) [inline website/content/tools/autoscaling/agent/high_availability.mdx:42]:
```suggestion
  to sync the leader when running in high availability mode. If the Nomad Autoscaler
```

File: website/content/tools/autoscaling/agent/high_availability.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  agent does not renew the lock within this period, the lock is considered lost.
+
+- `lock_delay` `(string: "30s")` - Defines the period that the lock will be blocked
+  after the TTL expired without being renewed. It is helps prevent split brain
```
Comment (pkazmierczak) [inline website/content/tools/autoscaling/agent/high_availability.mdx:46]:
```suggestion
  after the TTL expired without being renewed. This helps prevent split brain
```

File: website/content/tools/autoscaling/cli.mdx [hunk 4]
```diff
@@ -167,6 +167,26 @@ passed in via CLI arguments. The `agent` command accepts the following arguments
 - `-telemetry-circonus-broker-select-tag`: A tag which is used to select a broker
   ID when an explicit broker ID is not provided.
 
+- `-high-availability-enabled`: Specifies whether the Nomad Autoscaler agent will
+  start in high availability mode. If enabled, the agent instance attempts to
+  hold a lock over a Nomad variable and will only execute if the lock is
+  successfully acquired. It defaults to `false`.
+
+- `-high-availability-lock-path`: Defines the path of the Nomad variable that
+  will be used to sync the leader when running on high availability mode. This
```
Comment (pkazmierczak) [inline website/content/tools/autoscaling/cli.mdx:176]:
```suggestion
  will be used to sync the leader when running in high availability mode. This
```

File: website/content/tools/autoscaling/cli.mdx [hunk 5]
```diff
@@ -167,6 +167,26 @@ passed in via CLI arguments. The `agent` command accepts the following arguments
...
+  take part in the same leadership election process. Defaults to
+  `nomad-autoscaler/lock`.
+
+- `-high-availability-lock-ttl`: Defines the lease period or TTL of the lock used
+  to sync the leader when running on high availability mode. If the Nomad
+  Autoscaler agent does not renew the lock within this period, the lock is
+  considered lost. Defaults to `60s`.
+
+- `-high-availability-lock-delay`: Defines the period that the lock will be
+  blocked after the TTL expired without being renewed. It is helps prevent split
```
Comment (pkazmierczak) [inline website/content/tools/autoscaling/cli.mdx:187]:
```suggestion
  blocked after the TTL expired without being renewed. This helps prevent split
```

**Pull request merged**